src/datasets/arrow_dataset.py

```diff
@@ -1837,7 +1850,15 @@ def _getitem(self, key: Union[int, slice, str], decoded: bool = True, **kwargs)
         )
         return formatted_output
 
-    def __getitem__(self, key: Union[int, slice, str]) -> Union[Dict, List]:
+    @overload
+    def __getitem__(self, key: Union[int, slice, Iterable[int], Iterable[bool]]) -> Dict:  # noqa: F811
```

Comment:
We don't support lists of booleans, they are converted to integers.
```suggestion
    def __getitem__(self, key: Union[int, slice, Iterable[int]]) -> Dict:  # noqa: F811
```